softvis3d-frontend/src/components/scene/visualization/ThreeSceneService.ts

```diff
@@ -0,0 +1,112 @@
+///
+/// softvis3d-frontend
+/// Copyright (C) 2016 Stefan Rinderle and Yvo Niedrich
+/// stefan@rinderle.info / yvo.niedrich@gmail.com
+///
+/// This program is free software; you can redistribute it and/or
+/// modify it under the terms of the GNU Lesser General Public
+/// License as published by the Free Software Foundation; either
+/// version 3 of the License, or (at your option) any later version.
+///
+/// This program is distributed in the hope that it will be useful,
+/// but WITHOUT ANY WARRANTY; without even the implied warranty of
+/// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+/// Lesser General Public License for more details.
+///
+/// You should have received a copy of the GNU Lesser General Public
+/// License along with this program; if not, write to the Free Software
+/// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+///
+import {Vector3} from "three";
+import {SoftVis3dShape} from "../domain/SoftVis3dShape";
+import {SelectionCalculator} from "./SelectionCalculator";
+import {HtmlDom, Offset} from "../../../services/HtmlDom";
+import SoftVis3dScene from "./scene/SoftVis3dScene";
+import {Wrangler} from "./objects/Wrangler";
+
+export default class ThreeSceneService {
+
+    private threeScene: SoftVis3dScene;
+    private wrangler: Wrangler;
+
+    public constructor(softvis3dScene?: SoftVis3dScene, wrangler?: Wrangler) {
```

Comment:
Unit testing here was hard because i could not find a way to mock the constructors of SoftVis3dScene and Wrangler. To be able to mock the classes, i had to extend the constructor. Not nice, but at least it works.